### ISSUE TITLE
Do not wrap a multiline spread operator expression `*`

### ISSUE THREAD

**Issue by paul-dingemans** [pinterest/ktlint#2188]
See https://kotlinlang.org/spec/expressions.html#spread-operator-expressions for `*` operator.

Given code below:
```
fun foo(vararg items: String): Array<String> = arrayOf(*items)

fun bar(vararg items: String) = "bar"

val bar = bar(
    *foo(
        "a",
        "b"
    )
)
```
then it is formatted like:
```
fun foo(vararg items: String): Array<String> = arrayOf(*items)

fun bar(vararg items: String) = "bar"

val bar =
    bar(
        *
            foo(
                "a",
                "b",
            ),
    )
```
while it should have been kept unchanged. There is no sense in wrapping after the operator.

**Comment by FooIbar** [comment]
Regression on the latest snapshot build `ktlint-cli-1.0.1-20231013.062913-18`.

```
val foo =
    arrayOf(
        *items,
    )
```
Output: Line must not begin with "*" (standard:chain-wrapping)

**Comment by paul-dingemans** [comment]
> Regression on the latest snapshot build `ktlint-cli-1.0.1-20231013.062913-18`.
> 
> ```
> val foo =
>     arrayOf(
>         *items,
>     )
> ```
> 
> Output: Line must not begin with "*" (standard:chain-wrapping)

Can you please try again? I fixed this problem earlier this morning in PR #2306 (snapshot version [ktlint-cli-1.0.1-20231013.082432-20](https://oss.sonatype.org/content/repositories/snapshots/com/pinterest/ktlint/ktlint-cli/1.0.1-SNAPSHOT/ktlint-cli-1.0.1-20231013.082432-20-all.jar) and it seems that you are using a snapshot which was published a earlier today in which the problem actually existed.

**Comment by FooIbar** [comment]
> > Regression on the latest snapshot build `ktlint-cli-1.0.1-20231013.062913-18`.
> > ```
> > val foo =
> >     arrayOf(
> >         *items,
> >     )
> > ```
> > 
> > 
> > Output: Line must not begin with "*" (standard:chain-wrapping)
> 
> Can you please try again? I fixed this problem earlier this morning in PR #2306 (snapshot version [ktlint-cli-1.0.1-20231013.082432-20](https://oss.sonatype.org/content/repositories/snapshots/com/pinterest/ktlint/ktlint-cli/1.0.1-SNAPSHOT/ktlint-cli-1.0.1-20231013.082432-20-all.jar) and it seems that you are using a snapshot which was published a earlier today in which the problem actually existed.

Can confirm it's fixed in ktlint-cli-1.0.1-20231013.082432-20.
Thank you!
